src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styleObject = {};
+
+  const semicolon = ';';
+  const colon = ':';
+
+  const convertedToObject = sourceString
+    .split(semicolon)
+    .reduce((styleAccumulator, declaration) => {
+      const [property, value] = declaration.trim().split(colon);
+
+      if (value) {
+        styleAccumulator[property.trim()] = value.trim();
+      }
+
+      return styleAccumulator;
```

Comment:
[CODE KNOWLEDGE] - There is no need to make a copy before each operation. The `reduce` method does not mutate the original array, so you can omit the `styleObject` and use `{}` directly as the initial value for the accumulator.